Connection.php

```diff
@@ -130,4 +144,134 @@ public function flushDB($async = null)
+    {
+        $this->open();
+        $responses = [];
+        foreach ($commands as $i => $command) {
+            $responses[$i] = $this->sendRawCommand($command);
+        }
+        return $responses;
+    }
+
+    /**
+     * Execute Redis MULTI command with retry and retry interval.
+     * No Exception is thrown in case there is a connection error.
+     *
+     * @param array $commands - Command (multi dimensional array of commands).
+     *
+     * @return mixed
+     */
+    protected function retryExecuteMultiCommand(array $commands = [])
+    {
+        $tries = $this->retries;
+        while ($tries-- > 0) {
+            try {
+                // Try to open a Redis connection and execute the command.
+                return $this->executeMultiCommandOnce($commands);
+            } catch (RedisException $exception) {
+                // Log any exception with Yii error.
+                Yii::error($exception, __METHOD__);
+                // In case Redis is not accessible, close the connection, wait for the retry interval.
+                $this->close();
+                if ($this->retryInterval > 0) {
+                    usleep($this->retryInterval);
+                }
+            }
+        }
+    }
+
+    /**
+     * Execute Redis command.
+     *
+     * @param array $command - Command to run on Redis.
+     * 
+     * @return mixed
+     */
+    public function executeCommand(array $command = [])
+    {
+        // Run Redis command with retries.
+        if ($this->retries > 0) {
+            return $this->retryExecuteCommand($command);
+        }
+
+        return $this->executeCommandOnce($command);
+    }
+
+    /**
+     * Execute Redis command once. Will throw an Redis Exception in case there is a connection error.
+     *
+     * @param array $command - Command to run on Redis.
+     * 
+     * @return mixed
+     */
+    protected function executeCommandOnce(array $command = [])
+    {
+        $this->open();
+        return $this->sendRawCommand($command);
+    }
+
+    /**
+     * Execute Redis command with retry and retry interval. No Exception is thrown in case there is a connection error.
+     *
+     * @param array $command - Command.
+     *
+     * @return mixed
+     */
+    protected function retryExecuteCommand(array $command = [])
+    {
+        $tries = $this->retries;
+        while ($tries-- > 0) {
+            try {
+                // Try to open a Redis connection and execute the command.
+                return $this->executeCommandOnce($command);
+            } catch (RedisException $exception) {
+                // Log any exception with Yii error.
+                Yii::error($exception, __METHOD__);
+                // In case Redis is not accessible, close the connection, wait for the retry interval.
+                $this->close();
+                if ($this->retryInterval > 0) {
+                    usleep($this->retryInterval);
```

Comment:
**question:** `usleep` expects the delay to be expressed in microseconds, so don't we need to pass the retry interval as `$this->retryInterval * 1000` here?

Reply:
Correct @christiandewitt ! I left it in microseconds while I was testing it on local not to wait too much for the call and I forgot to add it back. Pushing the changes now